src/interfaces/state-interface.ts

```diff
@@ -0,0 +1,42 @@
+import { EntityStatus, CorpTypeCd, FilingStatus } from '@/enums'
+import { DirectorIF, FilingDataIF, LedgerIF, OfficeAddressIF } from '@/interfaces'
+
+export interface StateIF {
+  // tombstone data
+  keycloakRoles: Array<string>
+  authRoles: Array<string>
+  username: string
+  currentDate: string // 'today' as YYYY-MM-DD
+
+  // entity info
+  entityName: string
+  entityType: CorpTypeCd
+  entityStatus: EntityStatus
+  entityBusinessNo: string
+  entityIncNo: string
+  entityFoundingDate: string
+  businessEmail: string
+  businessPhone: string
+  lastAnnualReportDate: string
+  lastFilingDate: Date
+  lastCoaFilingDate: Date
+  lastCodFilingDate: Date
```

Comment:
Note that these are Javascript Dates. Until now we have mostly being using strings. Just be careful with using dates in this project.